openshift/machine-config-operator
# Bug 2104978: fix degraded pool state message

**- What I did**

when a node entered a degraded state, exited that specific state and then re-entered it later the MCO failed to update the pool degraded reason to the proper one. This was due to a missing annotation check in the updateNode function leading to a state where an update was not enqueued when the daemon state reason annotation changed.

fix `updateNode` to check for changes to the `MachineConfigDaemonReasonAnnotationKey`

also added extra logging for clarity.

**- How to verify it**
1) edit a file on a node
2) fix that error
3) edit a second file
4) edit the first file again, the degraded reason should be due to the first file not the second

**- Description for the changelog**
fix `updateNode` to check for changes to the `MachineConfigDaemonReasonAnnotationKey` this ensures that an update will be enqueued when the degraded reason changes. 


Signed-off-by: Charlie Doern <cdoern@redhat.com>
